@navikt/ds-react/accordion-menu/src/Context.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { createContext, useContext, useReducer, Dispatch } from "react";
+interface Props {
+  children: JSX.Element | JSX.Element[];
+}
+
+export interface Store {
+  activeAnchor?: string;
+  anchors?: Anchor[];
+}
+
+export interface Anchor {
+  id: string;
+}
+
+export type Action =
+  | {
+      type: "INSERT_ANCHOR";
+      id: string;
+    }
+  | {
+      type: "REMOVE_ANCHOR";
+      id: string;
+    }
+  | {
+      type: "SET_ACTIVE_ANCHOR";
+      id: string;
+    };
+
+const initialState = {
+  anchors: [],
+};
+
+export const reducer = (state: Store, action: Action) => {
+  switch (action.type) {
+    case "INSERT_ANCHOR":
+      return {
+        ...state,
+        anchors: [...state.anchors, { id: action.id }],
+      };
+    case "REMOVE_ANCHOR":
+      return {
+        ...state,
+        anchors: state.anchors.filter((anchor) => anchor.id !== action.id),
+      };
+    case "SET_ACTIVE_ANCHOR":
+      return {
+        ...state,
+        activeAnchor: action.id,
+      };
+    default:
+      return state;
+  }
+};
+
+export const StoreContext = createContext({} as [Store, Dispatch<Action>]);
+export const StoreProvider = (props: Props) => {
+  const { children } = props;
+
+  return (
+    <StoreContext.Provider value={useReducer(reducer, initialState)}>
```

Comment:
Jeg synes `useReducer` er litt overkill her. Personlig ville jeg nok implementert dette med to `useState` (en for active, og en for arrayet), og eksponert tre funksjoner: `registerAnchor`, `unregisterAnchor`, og `setActiveAnchor`.

Reply:
Men det er jo smak og behag :)

Reply:
Fiksa :)